src/Glpi/Cache/CacheStorageFactory.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2018 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Cache;
+
+use Zend\Cache\StorageFactory;
+
+/**
+ * Glpi cache storage factory.
+ *
+ * @since 10.0.0
+ */
+class CacheStorageFactory extends StorageFactory
+{
+    private static $cacheDir;
+
+    public function __construct(string $cacheDir)
+    {
+        self::$cacheDir = $cacheDir;
+    }
+
+    public static function factory($cfg)
+    {
+        // Compute prefered adapter if 'auto' value or no value is used
+        if (!isset($cfg['adapter']) || 'auto' === $cfg['adapter']) {
+            if (function_exists('apcu_fetch')) {
+                $cfg['adapter'] = 'apcu';
+            } else if (function_exists('wincache_ucache_add')) {
```

Comment:
since apcu may be present on windows but is known not to work properly; maybe should we check wincache first